config/file_config.go

```diff
@@ -348,6 +349,9 @@ func (c CollectionConfig) GetIncomingQueueSize() int {
 // GetNumCollectLoops returns the number of parallel collection loops.
 // Ensures the value is at least 1.
 func (c CollectionConfig) GetNumCollectLoops() int {
+	if c.NumCollectLoops == 0 {
+		return runtime.GOMAXPROCS(0)
```

Comment:
I don't know if this is a valid case. What if runtime.GOMAXPROCS return 0? Maybe it's still worth to do the max operation when we are taking the GOMAXPROCS value